x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/save_custom_dashboard.ts

```diff
@@ -42,33 +43,32 @@ export function initSaveCustomDashboardRoute(framework: KibanaFramework) {
         ...request.body,
       };
 
-      if (!payload?.id) {
-        const savedCustomDashboard = await savedObjectsClient.create<InfraCustomDashboard>(
-          INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE,
-          {
-            assetType: request.params.assetType,
-            ...payload,
-          }
-        );
+      const { assetType } = request.params;
 
-        return response.ok({
-          body: InfraSaveCustomDashboardsResponseBodyRT.encode(savedCustomDashboard.attributes),
+      const customDashboards = await findCustomDashboard(assetType, savedObjectsClient);
+
+      const dashboardExist = customDashboards.find(
+        (customDashboard) =>
+          customDashboard.dashboardSavedObjectId === payload.dashboardSavedObjectId
+      );
+
+      if (dashboardExist) {
+        return response.badRequest({
+          body: `Custom dashboard for ${assetType} with id ${payload.dashboardSavedObjectId} already exist`,
```

Comment:
Nit: the message is a bit confusing to me because the id here refers to the dashboard id, not the custom dashboard SO. 

Perhaps something like `Dashboard with id ${id} has already been linked to ${assetType}`?

Reply:
Good idea, thank you! I changed the message based on the suggestion :)